### DISCUSSION TITLE
How We Classify Banks

### DISCUSSION THREAD

**Discussion by jamcat22** [2factorauth/twofactorauth#6168] [General]
Currently, we classify a number of banks as having 2FA, when they only require it for certain transactions, not for logging in to their website. My question is whether or not we should change any of the banks listed below to `tfa: No`, add an exception to each of them, or find a way to include a `transactional` flag on the website.

Below is a list of banks that do not currently fall under our definition of 2FA, despite having the `tfa` flag set to `Yes`. I'm curious as to how everyone feels we should handle these types of situations with these banks and any additional ones going forward.
- Barclays UK's PINSentry doesn't qualify for our definition of 2FA, as their [website](http://www.barclays.co.uk/Helpsupport/UpgradetoPINsentry/P1242559314766) shows that it is only required for certain transactions.
- Citibank Australia's OTP doesn't qualify for our definition of 2FA, as it seems that they only require it for certain transactions. Their [website](http://www.citibank.com.au/otp/) says:
  ![You'll need to enter this anytime you want to perform select online transactions or query through Citibank Online.](https://cloud.githubusercontent.com/assets/1897053/15996479/f922f71c-30e4-11e6-8461-2d5fe503aff5.JPG)
- Commonwealth Bank of Australia's NetCode doesn't qualify for our definition of 2FA, as their [website](https://www.commbank.com.au/security-privacy/netcode.html) states that it's only required for certain transactions.
- first direct's Secure Key doesn't qualify for our definition of 2FA, as it can be bypassed using a password to login. Their [website](http://www1.firstdirect.com/1/2/securekey/options) says:
  ![...this doesn't mean you can't access your accounts if you don't have it with you. You can still log on without it and have limited access to Internet Banking by selecting the link at the top of the log on page...](https://cloud.githubusercontent.com/assets/1897053/15994744/e4b6055a-30c9-11e6-8b56-9d38756f6184.JPG)
- HSBC's Secure Key doesn't qualify for our definition of 2FA, as their [website](https://www.hsbc.co.uk/1/2/customer-support/online-banking-security/secure-key#demos) shows how you can choose the "Without Secure Key" tab when logging in to bypass 2FA.
- Nationwide Building Society's card reader doesn't qualify for our definition of 2FA, as the it can be bypassed using a password to login. Their [website](http://www.nationwide.co.uk/support/security-centre/internet-banking-security/card-reader-and-security-questions#xtab:what-is-a-card-reader) says:

> You can still log in with your memorable data and passnumber, but by using your card reader you may reduce the number of times it's needed to confirm your online transactions.
- Natwest's card reader doesn't qualify for our definition of 2FA, as they only require the card reader be used for certain transactions. In fact, their [website](http://personal.natwest.com/global/security-centre/stay-safe-online.html) specifically says:
  ![We will never ask you to use your card reader to log in to Online Banking, and we will never phone you to ask for your card reader details.](https://cloud.githubusercontent.com/assets/1897053/15994664/cadee716-30c7-11e6-8e99-727a25116bed.JPG)
- Santander's SMS-based OTP doesn't qualify for our definition of 2FA, as they only require an OTP for certain transactions. Their [website](http://www.santander.co.uk/uk/help-support/security-centre) says: 
![...we send these unique codes to your mobile to security check payments that you have recently set up (and) requests you make to amend some important details like your address.](https://cloud.githubusercontent.com/assets/1897053/21755201/1c52d44a-d5cd-11e6-9f75-d572ce4c17bd.png)

- State Bank of India's OTP application doesn't qualify for our definition of 2FA, as they only require an OTP for certain transactions.

**Comment by mxxcon** [comment]
Yup, I say all of these updated as `tfa: no`.
And perhaps add faq section that elaborates on differences between authentication and authorization?

**Comment by zD12** [comment]
You're right. These don't qualify as two-factor.

**Comment by jimsug** [comment]
I've had a look at the list of banks... seems very hit and miss, and the vast majority (but certainly not all) utilise some kind of step-up authentication and that we'll have a long, long list of banks that have this. I suspect that without further explanation, attempts to petition these banks to change it will fall on deaf ears. Their staff aren't trained on the difference, or why it matters (and they'll probably just say "we _do_ use two-factor authentication".

Any thoughts on adding adding a `stepup` property? This would indicate that the service requires a second factor only for certain actions, and perhaps with a link to further information on the difference ([this](https://www.quora.com/What-is-a-step-up-authentication/answer/John-Haggard-3) Quora answer actually does a decent job, IMO), and why two-factor authentication is preferable to this.

**Comment by indolering** [comment]
I disagree that not requiring 2FA in some scenarios "doesn't qualify as two factor":

* Allowing low risk actives without 2FA reduces user inconvenience and boosts adoption of 2FA.  Not everyone has a hardware token on each machine and multiple backups.  And even with those enabled, my GF complains about all the 2FA I have forced on her accounts.
* Requiring 2FA for high risk transactions could be used to create a finer grained security model, as malware (etc) can't perform arbitrary transactions after logging in.

2FA shouldn't just be an improved `sudo` command with arbitrary access to an entire system, it's better if we scope capabilities to specific tasks.

Mandating *which* tasks is tricky and probably best figured out by a standards body, not an ad-hoc site run by volunteers.  That sort of granular detail is probably best expressed as a grading system, not a binary yes/no.

**Reply by mxxcon** [reply]
Perfect example of confusing authentication and authorization.
